### ISSUE TITLE
Application insights doesn't show failed request

### ISSUE THREAD

**Issue by Zenuka** [Azure/azure-functions-host#6511]
#### Investigative information
Please provide the following:

- Timestamp: 2020-08-17T06:45:10.850Z
- Function App version: v3
- Function App name: xxx
- Function name(s) (as appropriate): UnauthorizedTestTrigger
- Invocation ID: 34d91b83-abbf-4b2f-9ece-3a7f2ccef128
- Region: West Europe

#### Expected behavior
I would expect requests with a error status code (like 401) to be in the failed requests tab of Application Insights but they seem to be successful requests.

#### Actual behavior
Requests are marked as successful when they have an error status code.
![image](https://user-images.githubusercontent.com/5391012/90366843-1d3d5200-e068-11ea-8241-b4cff6490c8d.png)

#### Known workarounds
I'm not aware of any workarounds.

#### Related information 
See example below.
<details>
<summary>Source</summary>

```csharp
[FunctionName("UnauthorizedTestTrigger")]
public IActionResult Run(
    [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "UnauthorizedTestTrigger")] HttpRequest req
)
{
    return new UnauthorizedResult();
}
```
</details>

**Comment by Zenuka** [comment]
Any idea where this issue might be located? Do you think it's in this repository or another? Happy to take a stab at it.

**Comment by soninaren** [comment]
@Zenuka , Thank you for the offering to take a look at this. 

@brettsam any ideas here? Is this by design?

**Comment by brettsam** [comment]
Yes -- if the function completed successfully and did not throw an exception, we mark it as successful. We only consider a function failed if it did not execute correctly.

@mathewc -- I swear we had an issue to add functionality that would allow you to (for example) throw an `HttpResponseException` and we'd mark it as failed, but use the Status, etc, from the exception in the response. I just hunted around but couldn't find it. If it doesn't exist, we can repurpose this one.

**Comment by MartinWickman** [comment]
@brettsam A thrown exception is very well handled by app insights and gives lots of nice information for debugging and investigation. 

But the issue, just to clarify, is that a thrown exception **always**  generates a 500. Now let's say you want to *return* 502 to the client, but you *also* want app insight to register it in the same way as an ordinary exception. 

No way to do that as far as I know.

Just a though though: Maybe there could be something like a `FunctionFailedException (int httpCode)` we could throw to clean this up once and for all.

**Comment by JLuzz** [comment]
@brettsam can we get a way to throw custom status code on failed executions?

If I wrote the following code in a function app, I would expect a `401` to be the returned status
```
const error = new Error("test")
error.status = 401

throw error
```

In most frameworks this is a failed execution and the statusCode comes back as a `401`.
However, in Azure Function app runtime this is forced to a `500`

**Comment by doriansobacki** [comment]
Do you happen to have any updates on this one? Does anyone know any workarounds for the problem? We really would like to see 4xx to be reported as failures and not only unhandled exceptions. 

I get it that function apps are not web apps but then we should at least have control over what is reported as a failure and what is not
